hazelcast/src/main/java/com/hazelcast/query/impl/predicates/LikePredicate.java

```diff
@@ -43,6 +47,38 @@ public LikePredicate(String attributeName, String expression) {
         this.expression = expression;
     }
 
+    @Override
+    public Set<QueryableEntry> filter(QueryContext queryContext) {
+        Index index = queryContext.matchIndex(attributeName, QueryContext.IndexMatchHint.PREFER_ORDERED);
+        String indexPrefix = expression.substring(0, expression.length() - 1);
+        return index.getRecords(indexPrefix, true, indexPrefix + "\uFFFF", false);
+    }
+
+    @Override
+    public boolean isIndexed(QueryContext queryContext) {
+        Index index = queryContext.matchIndex(attributeName, QueryContext.IndexMatchHint.PREFER_ORDERED);
+        return index != null && index.isOrdered() && expressionCanBeUsedAsIndexPrefix();
+    }
+
+    private boolean expressionCanBeUsedAsIndexPrefix() {
+        boolean escape = false;
+        for (int i = 0; i < expression.length(); i++) {
+            char c = expression.charAt(i);
+            if (c == '\\') {
+                escape = true;
+            } else {
```

Comment:
What if I have an expression `"\\%"`. Then the `\` is actually escaping the second `\`? But the algorithm will consider like it is escaping the `%` symbol?

Reply:
good catch added tests and fixed